### PR TITLE
DCOS-37981 Just retrieve individual packages, instead of whole repo

### PR DESCRIPTION
**Pull request by nickbp**
- Use new endpoint to just download info for the targeted package, instead of all packages
- Just use urllib instead of `requests` -> `curl` fallback

### CODE REVIEW

File: tools/universe/package_manager.py [hunk 1]
```diff
@@ -9,129 +9,64 @@
...
 
 LOGGER = logging.getLogger(__name__)
 
 
 class PackageManager:
     """A simple package manager for retrieving universe packages"""
-    def __init__(self, universe_url="https://universe.mesosphere.com/repo",
-                 dcos_version="1.10",
+    def __init__(self,
+                 universe_package_prefix="https://universe.mesosphere.com/package/",
```
Comment (elezar) [inline tools/universe/package_manager.py]: When I run: 
```
curl -L --header "User-Agent: dcos/1.11" --header "Accept: application/vnd.dcos.universe.repo+json;charset=utf-8;version=v4" https://universe.mesosphere.com/package/
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>865E5C49FA467D26</RequestId><HostId>zel8PgJwmSQXkuFzhuoFpqaLX2F5dQ7ugbeB7snM7BtbMia45hbnZHZOd2t+j5xc0eHrYsy8fMw=</HostId></Error>%```

(whereas `repo` still returns data).

Does one need to be authenticated for this?

Comment (elezar) [reply tools/universe/package_manager.py]: OK. One also needs the package name.

File: tools/universe/package_manager.py [hunk 2]
```diff
@@ -9,129 +9,64 @@
...
-            self._get_packages = _get_packages_with_curl
 
     def get_package_versions(self, package_name):
         """Get all versions for a specified package"""
+        if self._dry_run:
+            return DryRunPackages()
+
+        if package_name not in self.__package_cache:
+            LOGGER.info("Retrieving information for package: %s", package_name)
+            url = self._universe_package_prefix + package_name
```
Comment (elezar) [inline tools/universe/package_manager.py]: We could use `urllib.parse.urljoin(self._universe_package_prefix, package_name)` instead?

Comment (nickbp) [reply tools/universe/package_manager.py]: Good point, done.

**Pull request merged**